kernel/src/hil/aes.rs

```diff
@@ -0,0 +1,14 @@
+use returncode::ReturnCode;
+
+pub trait AESDriver {
+    fn init(&self);
+    fn set_key(&self, key: &'static mut [u8], len: u8);
```

Comment:
Hmm wouldn't be a problem. As I see it there are three scenarios
- Key == 16 bytes  - use the key
- Key < 16 bytes    - append with random bytes until it's 16 bytes
- Key > 16 bytes    - slice 16 first bytes

But this can introduce problems if a symmetric key is shared between two parties and a typo or similar is entered in the userapp i.e. not 16 bytes. Then the encryption/decryption will not work. The best is maybe just returncall "INVALID KEYLEN" or something similar. Thoughts?